python/ray/util/spark/cluster_init.py

```diff
@@ -1851,10 +1889,21 @@ def start(
             worker_port_range_end,
         ) = _preallocate_ray_worker_port_range()
 
-        ray_head_node_cmd = [
-            sys.executable,
-            "-m",
-            "ray.util.spark.start_ray_node",
+        if uses_pex:
+            import ray
+            ray_path = ray.__path__
+            ray_head_node_cmd = [
+                os.getenv("RAY_PEX_FILE"),
+                f"{ray_path[0]}/util/spark/start_ray_node.py",
+            ]
+        else:
+            ray_head_node_cmd = [
+                sys.executable,
+                "-m",
+                "ray.util.spark.start_ray_node",
+            ]
+
+        ray_head_node_cmd.extend([
```

Comment:
Another case: autoscaling is disabled. You need to update that case Ray head node command.

Reply:
Ah I see your point, I will modify+test+push to handle that

Reply:
@WeichenXu123 please take a look again, I made sure to trigger both autoscaling and fixed number of workers after making the changes. 

![image](https://github.com/ray-project/ray/assets/31358000/aa0e6ea5-82ce-4178-82ec-14358031bd3b)

Reply:
I think this place you need to update it too:

https://github.com/ray-project/ray/blob/6e10ca79bb623a0004c8ad5300a3f5d8aa2c4eb3/python/ray/util/spark/cluster_init.py#L682

Does your test cover it ?